browser-extension/mv3/src/service-worker/services/ruleExecutionHandler.ts

```diff
@@ -3,14 +3,23 @@ import extensionIconManager from "./extensionIconManager";
 import { DataScope, TAB_SERVICE_DATA, tabService } from "./tabService";
 import rulesStorageService from "../../rulesStorageService";
 import { CLIENT_MESSAGES } from "common/constants";
+import { Variable, onVariableChange } from "../variable";
+import { isExtensionEnabled } from "./utils";
 
 interface RulesExecutionLog {
   ruleId: string;
   requestDetails: chrome.webRequest.WebRequestDetails | chrome.webRequest.WebResponseDetails;
 }
 
 class RuleExecutionHandler {
-  constructor() {}
+  private isExtensionEnabled = true;
+
+  constructor() {
+    isExtensionEnabled().then((status) => (this.isExtensionEnabled = status));
+    onVariableChange<boolean>(Variable.IS_EXTENSION_ENABLED, (currentStatus) => {
+      this.isExtensionEnabled = currentStatus;
+    });
+  }
 
```

Comment:
Instead of disabling it in RuleExecutionHandler, we shouldn't even capture the request for execution in `webRequestInterceptor`